webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/services/healthcheck/HealthcheckNotificationService.scala

```diff
@@ -0,0 +1,75 @@
+/*
+*************************************************************************************
+* Copyright 2020 Normation SAS
+*************************************************************************************
+*
+* This file is part of Rudder.
+*
+* Rudder is free software: you can redistribute it and/or modify
+* it under the terms of the GNU General Public License as published by
+* the Free Software Foundation, either version 3 of the License, or
+* (at your option) any later version.
+*
+* In accordance with the terms of section 7 (7. Additional Terms.) of
+* the GNU General Public License version 3, the copyright holders add
+* the following Additional permissions:
+* Notwithstanding to the terms of section 5 (5. Conveying Modified Source
+* Versions) and 6 (6. Conveying Non-Source Forms.) of the GNU General
+* Public License version 3, when you create a Related Module, this
+* Related Module is not considered as a part of the work and may be
+* distributed under the license agreement of your choice.
+* A "Related Module" means a set of sources files including their
+* documentation that, without modification of the Source Code, enables
+* supplementary functions or services in addition to those offered by
+* the Software.
+*
+* Rudder is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+* GNU General Public License for more details.
+*
+* You should have received a copy of the GNU General Public License
+* along with Rudder.  If not, see <http://www.gnu.org/licenses/>.
+*
+*************************************************************************************
+*/
+
+package com.normation.rudder.services.healthcheck
+import com.normation.errors.IOResult
+import com.normation.zio._
+import net.liftweb.common.Loggable
+import zio.Ref
+import zio._
+import zio.duration._
+
+
+class HealthcheckNotificationService(
+    healthcheckService: HealthcheckService
+  , schedulerPeriod   : Duration
+) {
+
+  val healthcheckCache : Ref[List[HealthcheckResult]] = (for {
+    v <- Ref.make[List[HealthcheckResult]](List.empty)
+  } yield v).runNow
+
+  private[this] def reloadCache(ref: Ref[List[HealthcheckResult]]) = {
+    for {
+      fiber <- (healthcheckService.runAll).fork
+      checks <- fiber.join
+      _ <- ref.set(checks)
+    } yield {
+      checks
+    }
+  }
+
+  def updateCacheFromExt(hc: List[HealthcheckResult])= {
+    for {
+      _ <- healthcheckCache.set(hc)
+      test <- healthcheckCache.get
+      _ <- IOResult.effect{println(s"Update Cache from api :${test}")}
+
+    } yield ()
+  }
+
+  reloadCache(healthcheckCache).repeat(Schedule.spaced(20.seconds))
```

Comment:
doesn't seems to reload the cache..

Reply:
you need to `forkDaemon` to make it runs on its own fiber  (else it will block, repeating for ever and forbiding the prog to go further), and to `runNow` to actually make it do something.

But put it in a  `runNow` method to make it clears that it goes out of the pure wonderland of FP.